here-naksha-lib-model/src/jvmTest/kotlin/naksha/model/PropertyFilterTest.kt

```diff
@@ -0,0 +1,161 @@
+package naksha.model
+
+import naksha.base.AnyObject
+import naksha.base.Int64
+import naksha.jbon.JbEncoder
+import naksha.model.objects.NakshaFeature
+import naksha.model.request.ExecutedOp
+import naksha.model.request.PropertyFilter
+import naksha.model.request.ReadFeatures
+import naksha.model.request.ResultTuple
+import naksha.model.request.query.*
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import kotlin.test.assertEquals
+
+class PropertyFilterTest {
+
+    companion object {
+        lateinit var resultTuple : ResultTuple
+        val nestedJson = AnyObject()
+
+        @JvmStatic
+        @BeforeAll
+        fun setupTuple(): Unit {
+            // create the feature
+            val feature = NakshaFeature()
+            feature.properties["foo"] = "bar"
+            feature.properties["number"] = 1.1
+            nestedJson["bool"] = true
+            nestedJson["nullProps"] = null
+            val innerJson = AnyObject()
+            innerJson["a"] = 1
+            nestedJson["array"] = arrayOf("one", "two", "three", innerJson)
+            feature.properties["json"] = nestedJson
+            // build tuple containing the feature
+            val encoder = JbEncoder()
+            val byteArray = encoder.buildFeatureFromMap(feature)
+            val storeNumber = StoreNumber(0, Int64(0))
+            val version = Version(0)
+            val flag = Flags(0)
+            val tupleNumber = TupleNumber(
+                storeNumber = storeNumber,
+                uid = 0,
+                version = version,
+            )
+            val mockStorage = mock<IStorage>()
+            val tuple = Tuple(
+                storage = mockStorage,
+                tupleNumber = tupleNumber,
+                Metadata(
+                    storeNumber = storeNumber,
+                    updatedAt = Int64(0),
+                    uid = 0,
+                    id = "",
+                    appId = "",
+                    author = null,
+                    version = version,
+                    type = null,
+                    flags = flag,
+                ),
+                byteArray
+            )
+            resultTuple = ResultTuple(
+                storage = mockStorage,
+                tupleNumber = tupleNumber,
+                op = ExecutedOp.READ,
+                featureId = null,
+                tuple = tuple
+            )
+        }
+    }
+
+    @Test
+    fun testFilter() {
```

Comment:
This test is way to broad - single test should cover single scenario. This is not just an academic rule, it's something that will make our lifes easier in the long run. 

Let me show an example of what I mean by that: if this test fails, for example on CI/CD, and we would get an email or whatever saying "testFilter() failed for pipeline" - would you immediately know what was the issue? I know I wouldn't - even if I examined the logs of "assertEquals" - there's no way for me to know what really failed without opening this test and possibly starting a debug session.

Imagine this failure would be encountered by someone new to Naksha - this multiplies the time needed to properly identify the issue even more.

This test is a mix of many different scenarios
- it tests for both happy and unhappy paths (sometimes you expect `null`, sometimes you expect `resultTuple`)
- it tests for different `pQuery` types
- it tests for different `AnyOp` types when we have generic `PQuery` to be handled

Each of the above should not even be different test but even "family" of tests.
So instead of `testFilter` I would expect individual tests for:
- returning tuple when `pQuery` is null
- each of the `IPropertyQuery` types
- each of `AnyOp` types that we cover
- what happens when we encounter unkown `IProertyQuery` or `AnyOp`
etc
- different tests for successful and unsuccessful scenario ("shouldFailWhenX", "shouldSuccedWhenY")

Good news is, you have a lot of stuff done, it just needs being placed in different methods.